group-generators/helpers/data-providers/alchemy/index.ts

```diff
@@ -191,14 +192,14 @@ export class AlchemyProvider {
   }
 
   /**
-   *Fetches owners of NFTs with a matching tokenid
+   *Fetches owners of NFTs with a matching tokenid. This will take a long time if there are many NFTs
    * @param {Object} params - The parameters for fetching owners by token IDs.
    * @param {string} contractAddress - The address of the contract to query.
    * @param {string} chain - The blockchain network to query.
    * @param {string[]} tokenIds - An array of ERC-1155 token IDs to fetch owners for.
    * @returns {Promise<FetchedData>} - A promise that resolves to an object mapping owner addresses to the number of token balances they hold.
    */
-  private async getOwnersByTokenIds({
+  public async getOwnersByTokenIds({
```

Comment:
I also changed this back to being public so I can use it for this query. 
There are now 2 different queries that fetch owners of NFT collection based on tokenId. As mentioned before, they query the endpoints differently
1. `getOwnersByTokenIds()` - queries very verbose output and filters on it. Can accept multiple tokenIds.
2. `getOwnersForOneTokenId()` - queries 1 tokenId for owners. Good for ERC1155 where there can be 50000 owners of 1 tokenid

Reply:
Okay thanks for the info!
I think it could be even better to integrate the `getOwnersForOneTokenId` in the `getOwnersByTokenIds` (and set it as private).
So in the `getOwnersByTokenIds` we would test if only 1 tokenId has been provided: use the `getOwnersForOneTokenId`.

The UX will be way better for factory users (2 different functions can be disappointing)

Reply:
@sampolgar I don't know if ou see this comment so I ping you to it.
(You will have to change the interface schema etc... accordingly)